src/openrct2/peep/Peep.cpp

```diff
@@ -7182,6 +7182,16 @@ static void peep_update_walking(rct_peep * peep)
             litter_create(x, y, peep->z, direction, bp);
         }
     }
+    
+    rct_tile_element * path_element = map_get_path_element_at(peep->destination_x / 32, peep->destination_y / 32, peep->next_z);
+    if (path_element && path_element->flags & TILE_ELEMENT_FLAG_BLOCKED_BY_VEHICLE)
+    {
+        if (!(peep->x >> 5 == peep->destination_x >> 5 &&
```

Comment:
What are the lower 5 bits? Can they differ? In the line above you also divide `destination_[xy]` by 32 already, so you could just reuse the value here.

Reply:
These are "big"/"sprite" coordinates, which are 32 times as big as regular coordinates. This code checks if both refer to the same tile, rather than the exact same spot within that tile.

Reply:
Indeed. I suppose it depends on coding conventions whether to re-use,, I usually prefer to leave it to the compiler to do optimisations on that scale.

Reply:
@Gymnasiast unless `(x & 0x1f) != (destination_x & 0x1f)`, `x >> 5 == destination_x >> 5` implies `x == destination_x` and you've just wasted 4 instructions shifting things for no reason. Hence my question:

> What are the lower 5 bits? Can they differ?

@JeroenDStout leaving it to the compiler is absolutely fine, but a) you make it harder for the compiler to deduce the meaning of it by using `/` and `>>` b) you still pay the full price in unoptimised builds

Reply:
@janisozaur That is a very good point!